src/rules/jsx_no_comment_text_nodes.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use super::{Context, LintRule};
+use crate::handler::{Handler, Traverse};
+use crate::Program;
+use deno_ast::view::JSXText;
+use deno_ast::SourceRanged;
+
+#[derive(Debug)]
+pub struct JSXNoCommentTextNodes;
+
+const CODE: &str = "jsx-no-comment-text-nodes";
+
+impl LintRule for JSXNoCommentTextNodes {
+  fn tags(&self) -> &'static [&'static str] {
+    &["react", "jsx", "fresh"]
+  }
+
+  fn code(&self) -> &'static str {
+    CODE
+  }
+
+  fn lint_program_with_ast_view(
+    &self,
+    context: &mut Context,
+    program: Program,
+  ) {
+    JSXNoCommentTextNodesHandler.traverse(program, context);
+  }
+
+  #[cfg(feature = "docs")]
+  fn docs(&self) -> &'static str {
+    include_str!("../../docs/rules/jsx_no_comment_text_nodes.md")
+  }
+}
+
+const MESSAGE: &str =
+  "Comments inside children should be placed inside curly braces";
+
+struct JSXNoCommentTextNodesHandler;
+
+impl Handler for JSXNoCommentTextNodesHandler {
+  fn jsx_text(&mut self, node: &JSXText, ctx: &mut Context) {
+    //
+    let value = &node.inner.value;
+    if value.starts_with("//") || value.starts_with("/*") {
```

Comment:
Should these be stricter checks? Eg. ensuring there are only single lines or for block comments that it ends with `*/`?

Reply:
This is too difficult to do, because of expression nodes. In practice this is imo good enough.

```jsx
a = <p>/* foo {text}</p>
```